File: src/app/shared/model/request/entryProductDTORequest.ts
import { DtEntryProductDTORequest } from './dtEntryProductDTORequest';

export interface EntryProductDTORequest{
    idProvider?:number;
    idWharehouse?:number;
    idEntryDocument?:number;
    details?:DtEntryProductDTORequest[];
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/dto/request/EntryProductDTORequest.java
package com.sys.botica.crce.pe.sys_botica.dto.request;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import lombok.Data;

@Data
public class EntryProductDTORequest {

	@NotNull
	private Long idProvider;
	
	@NotNull
	private Long idWharehouse;
	
	@NotNull
	private Long idEntryDocument;
	
	@NotNull
	@Valid
	List<DtEntryProductDTORequest> details;
}
